File: source/analysis/top_gdp/top_gdp_table.R
library(dplyr)

Main <- function() {
  instub  <- "output/derived/wb_clean"
  outstub <- "output/analysis/top_gdp"

  df <- read.csv(file.path(instub, "gdp_education.csv"))

  df <- df %>%
    select(Country.Name, GDP_2010) %>%
    arrange(-GDP_2010) %>% top_n(5)

  WriteTable("<tab:top_gdp>", df, file.path(outstub, "top_gdp.txt"))

  df_diff <- data.frame(
    Countries = paste(df$Country.Name[5], "-", df$Country.Name[1]),
    Gap       = df$GDP_2010[5] - df$GDP_2010[1]
  )

  WriteTable("<tab:top_gdp_gap>", df_diff, file.path(outstub, "top_gdp_gap.txt"))
}


WriteTable <- function(tag, df, out) {
  write.table(tag, row.names = FALSE, quote = FALSE,
              file = out, col.names = FALSE)
  write.table(df, row.names = FALSE, quote = FALSE,
              file = out, append = TRUE, col.names = FALSE, sep = "\t")
}


Main()

File: source/analysis/top_gdp/top_gdp_table.py
from pathlib import Path
import pandas as pd
from source.lib.JMSLab.autofill import AutoFill

def Main():
    in_dir  = Path("output/derived/wb_clean")
    out_dir = Path("output/analysis/top_gdp")

    df_gdp = pd.read_csv(in_dir / "gdp_education.csv")
    df_top_gdp = df_gdp.sort_values(by='GDP_2010', ascending=False).iloc[0:5]
    df_top_gdp = df_top_gdp[["Country Name", "GDP_2010"]]

    WriteTable("<tab:top_gdp>", df_top_gdp, Path(out_dir, "top_gdp.txt"))


    top_country = df_top_gdp["Country Name"].iloc[0]
    fifth_country = df_top_gdp["Country Name"].iloc[4]

    df_diff = pd.DataFrame({
        "Countries": [f"{fifth_country} - {top_country}"],
        "Gap": [df_top_gdp["GDP_2010"].iloc[4] - df_top_gdp["GDP_2010"].iloc[0]]
    })

    WriteTable("<tab:top_gdp_gap>", df_diff, Path(out_dir, "top_gdp_gap.txt"))

def WriteTable(tag, df, out):
    with open(out, "w") as outfile:
        outfile.write(tag + "\n")
        df.to_csv(outfile, index=False, header=False, mode="a", sep="\t")

if __name__ == '__main__':
    Main()
